main/kotlin/com/example/people_managment_system/service/SystemService.kt

```diff
@@ -0,0 +1,177 @@
+    Returns a String response if insufficient or erroneous data is entered. */
+
+    fun saveNewEmployee(companyName: String ,newEmployeeDto: EmployeeDto): String {
+       val newEmployee = Employee()
+       newEmployee.employeeID = newEmployeeDto.employeeID
+       newEmployee.employeeName = newEmployeeDto.employeeName
+       newEmployee.email = newEmployeeDto.email
+       newEmployee.salary = newEmployeeDto.salary
+       newEmployee.company = companyRepository.getCompanyByName(companyName)
+       newEmployee.manager = managerRepository.getManagerById(newEmployeeDto.managerID)
+       newEmployee.team = teamRepository.getTeamById(newEmployeeDto.teamID)
+       return try {
+           employeeRepository.save(newEmployee)
+           "The Employee has been successfully saved to PeopleMs"
+       } catch (e: DataIntegrityViolationException) {
+           throw DataIntegrityViolationException("Could not add the employee to your company. " +
+                   "Incorrect or insufficient data supplied.")
+       }
+    }
+
+/* Updates the Team of an employee.
+   Takes in the company name as a path variable to ensure the Team ID relates to the right company.
+   Throws an exception if the employee ID does not exist. */
+
+    fun updateTeam(companyName: String, relevantEmployeeId: Int, relevantTeamId: Int): EmployeeDto {
+       var relevantEmployee: Employee? = null
+       for (item in employeeRepository.findAll()) {
+           if (item.employeeID == relevantEmployeeId &&
+                   item.company?.companyName == companyName) {
+               relevantEmployee = item
+               relevantEmployee.team = teamRepository.getTeamById(relevantTeamId)
+               employeeRepository.save(relevantEmployee)
+           }
+       }
+       return relevantEmployee?.let { mapEmployeeEntityToEmployeeDto(it) }
+           ?: throw NoSuchElementException("Could not find an employee with that ID")
+    }
+
+/* Updates the Manager of an employee.
+   Takes in the company name as a path variable to ensure the Manager ID relates to the right company.
+   Throws an exception if the employee ID does not exist. */
+
+fun updateManager(companyName: String, relevantEmployeeId: Int, relevantManagerId: Int): EmployeeDto {
+   var relevantEmployee: Employee? = null
+   for (item in employeeRepository.findAll()) {
+       if (item.employeeID == relevantEmployeeId &&
+           item.company?.companyName == companyName) {
+           relevantEmployee = item
+           relevantEmployee.manager = managerRepository.getManagerById(relevantManagerId)
+           employeeRepository.save(relevantEmployee)
+       }
+   }
+   return relevantEmployee?.let { mapEmployeeEntityToEmployeeDto(it) }
+       ?: throw NoSuchElementException("Could not find an employee with that ID")
+}
+
+/* Deletes an employee's data.
+   Returns a String response if the employee ID does not exist. */
+
+    fun deleteEmployee(relevantEmployeeId: Int): String {

```

Comment:
When you move this to the EmployeeService class, you can call this method `delete`. Because, we know which class it is and it's a specific operation for Employee.

You can do similar to other classes such as Company.